entity-service/src/main/kotlin/com/egm/stellio/entity/service/EntityService.kt

```diff
@@ -96,7 +100,12 @@ class EntityService(
         }
 
         expandedEntity.properties.forEach { entry ->
-            createEntityProperty(entity, entry.key, entry.value)
+            if (entry.value.count { !it.containsKey(NGSILD_DATASET_ID_PROPERTY) } > 1)
+                throw BadRequestDataException("Property ${entry.key.extractShortTypeFromExpanded()} already has default instance")

```

Comment:
The message is not clear about the error. It should instead say that you can't have more than one default instance.

Reply:
To me it's still the same, you do processing but you could know that this would fail before just by cheking the payload.